src/Npgsql/NpgsqlConnection.cs

```diff
@@ -141,7 +142,15 @@ void GetPoolAndSettings()
             }
 
             // Connection string hasn't been seen before. Parse it.
-            Settings = new NpgsqlConnectionStringBuilder(_connectionString);
+            if (Regex.IsMatch(_connectionString, @"^\w+://"))
```

Comment:
@y-code In #2512 I used the following check: https://github.com/npgsql/npgsql/pull/2512/files#diff-6daf78c3f8b7fabad257e702ed1ef44eR186 - do you think it makes sense to use this here?

Reply:
@jgilchrist Thank you for reviewing my PR. Yes, I tried System.Uri at first stage, and recognized that there's several problems with it to cover all types of PostgreSQL connection URIs that are documented in [PostgreSQL's documentation](https://www.postgresql.org/docs/current/libpq-connect.html#id-1.7.3.8.3.3). I've listed up the problems with System.Uri in `Notes` section of my first comment in this PR, so please have a look at it.

Reply:
Ah, sorry, I missed that part of your notes! Makes sense - resolving.

Reply:
Ah! Sorry, I missed that part of your notes. Thanks - makes sense.